src/scheduler/check.cpp

```diff
@@ -1570,13 +1570,12 @@ check_normal_node_path(status *policy, server_info *sinfo, queue_info *qinfo, re
 			/* if there are nodes assigned to the queue, then check those */
 			if (qinfo->has_nodes)
 				ninfo_arr = qinfo->nodes;
-			else if (sinfo->svr_to_psets != NULL) {	/* Multi-server psets */
-				int i;
-
+			else if (!(sinfo->svr_to_psets.empty())) {	/* Multi-server psets */
 				/* Find the pset of the server which owns the job */
-				for (i = 0; sinfo->svr_to_psets[i] != NULL; i++) {
-					if (strcmp(sinfo->svr_to_psets[i]->svr_inst_id, resresv->job->svr_inst_id) == 0) {
-						msvr_pset[0] = sinfo->svr_to_psets[i]->np;
+				for (auto spset = sinfo->svr_to_psets.begin();
```

Comment:
you can simplify it to just say "for (auto spset: sinfo->svr_to_psets)" instead.